components/input/NumericInput.js

```diff
@@ -0,0 +1,132 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { IconChevronUpSmallOutline, IconChevronDownSmallOutline } from '@teamleader/ui-icons';
+import Icon from '../icon';
+import InputBase from './InputBase';
+import theme from './theme.css';
+
+class SpinnerControls extends PureComponent {
+  render() {
+    const { inverse, handleOnSpinnerUpClick, handleOnSpinnerdownClick } = this.props;
+    const iconProps = {
+      color: inverse ? 'teal' : 'neutral',
+      element: 'button',
+      tint: inverse ? 'lightest' : 'darkest',
+    };
+
+    return (
+      <div className={theme['spinner']}>
+        <Icon className={theme['spinner-up']} onClick={() => handleOnSpinnerUpClick()} {...iconProps}>
+          <IconChevronUpSmallOutline />
+        </Icon>
+        <Icon className={theme['spinner-down']} onClick={() => handleOnSpinnerdownClick()} {...iconProps}>
+          <IconChevronDownSmallOutline />
+        </Icon>
+      </div>
+    );
+  }
+}
+
+class NumericInput extends PureComponent {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.value !== undefined) {
+      const newValue = nextProps.value || '';
+      if (newValue !== prevState.value) {
+        return {
+          value: newValue,
+        };
+      }
+    }
+    return null;
+  }
+
+  state = {
+    value: '',
+  };
+
+  handleOnChange = event => {
+    this.setState({ value: this.parseValue(event.currentTarget.value) });
+  };
+
+  updateStep = n => {
+    const { step } = this.props;
+
+    const currentValue = this.toNumber(this.state.value || 0);
+    const newValue = this.parseValue(currentValue + step * n);
+
+    if (newValue !== currentValue) {
+      this.setState({ value: newValue });
+    }
+  };
+
+  parseValue = value => this.bindToMinMax(this.toNumber(value));
+
+  toNumber = number => {
+    let float = parseFloat(number);
+    if (isNaN(float) || !isFinite(float)) {
+      float = 0;
+    }
+    return float;
+  };
+
+  bindToMinMax = value => {
+    const { min, max } = this.props;
+    return Math.min(Math.max(value, min), max);
+  };
+
+  handleIncreaseValue = () => {
+    this.updateStep(1);
```

Comment:
Q: Shouldn't this step actually be variable? If you now want the NumericInput to increment by 5, shouldn't this be this.props.step?

Reply:
`step` _is_ a prop, but I see where the confusion comes from. The argument passed to the `updateStep` method here is more of a factor, by which `this.props.step` is multiplied before adding it to the value.

Something like this happens, with `n` being the argument passed to `updateStep`:
```
this.state.value + this.props.step * n
```

Reply:
Okay, but if you do `this.props.step * n` and `n` is always 1, then `* n` is just redundant, no?

Reply:
`n` is not always 1, in the `handleDecreaseValue` method it is -1.
So it is useful to toggle between increasing and decreasing the value.

Reply:
Aha ok, then I didn't look far enough, thanks for elaborating 👍